### PR TITLE
Migrate to hilt

### PR DESCRIPTION
**Pull request by Skrilltrax**

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -58,6 +59,9 @@ android {
 
     buildFeatures.viewBinding = true
     kapt.correctErrorTypes = true
+    lintOptions {
+        disable("InvalidFragmentVersionForActivityResult")
```
Comment (msfjarvis) [inline app/build.gradle.kts:63]: This is a legitimate lint pattern and almost never a good idea to suppress.

File: app/src/main/java/me/skrilltrax/themoviedb/di/NetworkModule.kt [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+@InstallIn(ApplicationComponent::class)
+@Module
+object NetworkModule {
+
+    @Provides
+    @Singleton
```
Comment (msfjarvis) [inline app/src/main/java/me/skrilltrax/themoviedb/di/NetworkModule.kt:22]: Shouldn't `ApplicationComponent` imply this?

File: app/src/main/java/me/skrilltrax/themoviedb/ui/homepage/tv/TVViewPagerFragment.kt [hunk 3]
```diff
@@ -67,11 +67,11 @@ class TVViewPagerFragment : Fragment(), ListItemClickListener {
...
     companion object {
         fun newInstance(fragmentType: Int): TVViewPagerFragment {
-            val bundle = Bundle()
-            val commonViewPagerFragment = TVViewPagerFragment()
-            bundle.putInt("fragmentType", fragmentType)
-            commonViewPagerFragment.arguments = bundle
-            return commonViewPagerFragment
+            return TVViewPagerFragment().apply {
+                val bundle = Bundle()
+                bundle.putInt("fragmentType", fragmentType)
```
Comment (msfjarvis) [inline app/src/main/java/me/skrilltrax/themoviedb/ui/homepage/tv/TVViewPagerFragment.kt:72]: Can use the `bundleOf` extension from androidx.core here

File: app/src/main/java/me/skrilltrax/themoviedb/ui/search/SearchActivity.kt [hunk 4]
```diff
@@ -16,14 +17,15 @@ import me.skrilltrax.themoviedb.adapter.MovieListAdapter
 import me.skrilltrax.themoviedb.interfaces.ListItemClickListener
 import me.skrilltrax.themoviedb.model.list.movie.MovieListResultItem
 import me.skrilltrax.themoviedb.network.api.movie.MovieApiInterface
-import org.koin.android.ext.android.inject
 import retrofit2.HttpException
 
 class SearchActivity : AppCompatActivity(), ListItemClickListener {
```
Comment (msfjarvis) [inline app/src/main/java/me/skrilltrax/themoviedb/ui/search/SearchActivity.kt:22]: Missing `@AndroidEntryPoint`

**Review by msfjarvis — COMMENTED** [review]
Minor problems, everything else looks solid.

File: app/build.gradle.kts (hunk 1 above)
Comment (Skrilltrax) [reply app/build.gradle.kts:63]: I have updated my `fragments-ktx` to `1.3.0-alpha08` but it was still showing this lint error so I'm guessing it's a false positive.

File: app/src/main/java/me/skrilltrax/themoviedb/ui/search/SearchActivity.kt (hunk 4 above)
Comment (Skrilltrax) [reply app/src/main/java/me/skrilltrax/themoviedb/ui/search/SearchActivity.kt:22]: This is useless atm, will fix though.

File: app/build.gradle.kts (hunk 1 above)
Comment (msfjarvis) [reply app/build.gradle.kts:63]: Try beta01

File: app/src/main/java/me/skrilltrax/themoviedb/di/NetworkModule.kt (hunk 2 above)
Comment (Skrilltrax) [reply app/src/main/java/me/skrilltrax/themoviedb/di/NetworkModule.kt:22]: ```Warning: A common misconception is that all bindings declared in a module will be scoped to the component the module is installed in. However, this isn’t true. Only bindings declarations annotated with a scope annotation will be scoped.```

This is from https://dagger.dev/hilt/components

File: app/build.gradle.kts (hunk 1 above)
Comment (Skrilltrax) [reply app/build.gradle.kts:63]: Did not help :(

Comment (msfjarvis) [reply app/build.gradle.kts:63]: Weird, I did the update in WireGuard and it worked https://github.com/WireGuard/wireguard-android/commit/9266487fe572e939170d58ab2a5c2a0e138ffbc1

Comment (Skrilltrax) [reply app/build.gradle.kts:63]: Yeah the problem is not that I'm not following the pattern but `ActivityResultFragmentVersionDetector.checkDslPropertyAssignment` accepts a parent parameter of type `String` but it is being passed as `null`.

**Pull request merged**